src/main/java/org/candlepin/subscriptions/resource/OptInResource.java

```diff
@@ -0,0 +1,96 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.resource;
+
+import org.candlepin.subscriptions.controller.OptInController;
+import org.candlepin.subscriptions.security.auth.AdminOnly;
+import org.candlepin.subscriptions.utilization.api.model.OptInConfig;
+import org.candlepin.subscriptions.utilization.api.resources.OptInApi;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.BadRequestException;
+
+/**
+ * Defines the API for opting an account and its org into the sync/reporting
+ * functionality.
+ */
+@Component
+@Profile("api")
+public class OptInResource implements OptInApi {
+
+    private OptInController controller;
+
+    @Autowired
+    public OptInResource(OptInController controller) {
+        this.controller = controller;
+    }
+
+    @AdminOnly
+    @Override
+    public void deleteOptInConfig() {
+        controller.optOut(validateAccountNumber(), validateOrgId());
+    }
+
+    @AdminOnly
+    @Override
+    public OptInConfig getOptInConfig() {
+        return controller.getOptInConfig(validateAccountNumber(), validateOrgId());
+    }
+
+    @AdminOnly
+    @Override
+    public OptInConfig putOptInConfig(Boolean enableTallySync, Boolean enableTallyReporting,
+        Boolean enableConduitSync) {
+        // NOTE: All query params are defaulted to 'true' by the API definition, however we
+        //       double check below.
+        return controller.optIn(
+            validateAccountNumber(),
+            validateOrgId(),
+            trueIfNull(enableTallySync),
+            trueIfNull(enableTallyReporting),
+            trueIfNull(enableConduitSync)
+        );
+    }
+
+    private String validateAccountNumber() {
+        String accountNumber = ResourceUtils.getAccountNumber();
+        if (accountNumber == null) {
+            throw new BadRequestException("Must specify an account number.");
+        }
+        return accountNumber;
+    }
+
+    private String validateOrgId() {
+        String ownerId = ResourceUtils.getOwnerId();
+        if (ownerId == null) {
+            throw new BadRequestException("Must specify an org ID.");
+        }
+        return ownerId;
+    }
+
+    private Boolean trueIfNull(Boolean toVerify) {
+        return Boolean.valueOf(toVerify == null);
```

Comment:
Saw some weird behavior where all flags were set to false, traced it to this method.

```suggestion
        if (toVerify == null) {
            return true;
        }
        return toVerify;
```
(or some such)

Reply:
@kahowell lol - Oh, my. That's brutal on my part. That's what I get for rushing I suppose. :/